Source/Csla.AspNetCore/Blazor/State/SessionManager.cs

```diff
@@ -77,15 +76,22 @@ private static void Replace(Session newSession, Session oldSession)
     /// the wasm client to the web server.
     /// </summary>
     /// <returns></returns>
-    public Task SendSession() => throw new NotSupportedException();
+    public Task SendSession()
+    {
+      var key = _sessionIdManager.GetSessionId();
+      var existingSession = _sessions[key];
```

Comment:
What about extracting this piece of code into a method
```csharp
private void UpdateSessionLastTouched(string key) {
  var existingSession = _sessions[key];
  if (existingSession != null)
    existingSession.LastTouched = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
}
```

This would also remove the lookup of the key twice :D
And use it here and in `GetSession()`

Reply:
Idea: Get fancy and add `internal void UpdateLastTouchedToNow()` method to the `Session` type and use that everywhere?